src/systems/systems/src/air_conditioning/cabin_air.rs

```diff
@@ -0,0 +1,697 @@
+            * zone_volume.get::<cubic_meter>()
+            / (Air::R * self.internal_air.temperature().get::<kelvin>());
+        let internal_energy = internal_mass
+            * Air::SPECIFIC_HEAT_CAPACITY_VOLUME
+            * self.internal_air.temperature().get::<kelvin>();
+
+        let equilibrium_temperature =
+            ((passenger_heat_energy + inlet_air_energy + inlet_door_air_energy
+                - outlet_air_energy
+                - outlet_door_air_energy
+                - wall_transfer_heat_energy)
+                * context.delta_as_secs_f64()
+                + internal_energy)
+                / (internal_mass * Air::SPECIFIC_HEAT_CAPACITY_VOLUME);
+        ThermodynamicTemperature::new::<kelvin>(equilibrium_temperature)
+    }
+
+    fn heat_transfer_through_wall_calculation(
+        &self,
+        context: &UpdateContext,
+        true_airspeed: Velocity,
+        zone_volume: Volume,
+    ) -> Power {
+        let external_convection_coefficient: f64 =
+            if true_airspeed < Velocity::new::<meter_per_second>(15.) {
+                self.natural_convection_coefficient_calculation(context)
+            } else {
+                self.forced_convection_coefficient_calculation(context, true_airspeed, zone_volume)
+            };
+        let internal_convection_coefficient: f64 =
+            self.natural_convection_coefficient_calculation(context);
+        let wall_specific_heat_transfer: f64 = (self.internal_air.temperature().get::<kelvin>()
+            - context.ambient_temperature().get::<kelvin>())
+            / (1. / internal_convection_coefficient
+                + CabinZone::FIBER_GLASS_BLANKET_THICKNESS
+                    / CabinZone::FIBER_GLASS_BLANKET_THERMAL_CONDUCTIVITY
+                + CabinZone::ALUMINIUM_ALLOW_THICKNESS
+                    / CabinZone::ALUMINIUM_ALLOY_THERMAL_CONDUCTIVITY
+                + 1. / external_convection_coefficient);
+        let zone_surface_area: f64 = 2.
+            * std::f64::consts::PI
+            * self
+                .characteristic_length_calculation(zone_volume)
+                .get::<meter>();
+        Power::new::<watt>(wall_specific_heat_transfer * zone_surface_area)
+    }
+
+    fn natural_convection_coefficient_calculation(&self, context: &UpdateContext) -> f64 {
+        // Temperature differential should be based on the wall temperature instead of the inside vs outside. This is a simplification
+        let temperature_differential: f64 = self.internal_air.temperature().get::<kelvin>()
+            - context.ambient_temperature().get::<kelvin>(); // Kelvin
+
+        // Convection coefficient for horizontal cylinder in air
+        let convection_coefficient: f64 =
+            Self::CONVECTION_COEFFICIENT_CONSTANT_FOR_NATURAL_CONVECTION
+                * (temperature_differential.abs() / Self::A320_CABIN_DIAMETER).powf(1. / 4.);
+        convection_coefficient
+    }
```

Comment:
Unfortunately, addition and subtraction are not implemented for the `ThermodynamicTemperature` type. I'm assuming this is because of the different temperature scales, for example between Celsius and Fahrenheit, where subtracting in one unit and then converting is not the same as converting first and then subtracting.

https://docs.rs/uom/0.30.0/uom/si/thermodynamic_temperature/index.html